lib/PuppeteerSharp/Helpers/TempDirectory.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using PathHelper = System.IO.Path;
+
+namespace PuppeteerSharp.Helpers
+{
+    /// <summary>
+    /// Represents a directory that is deleted on disposal.
+    /// </summary>
+    internal class TempDirectory : IDisposable
+    {
+        private Task _deleteTask;
+
+        public TempDirectory()
+            : this(PathHelper.Combine(PathHelper.GetTempPath(), PathHelper.GetRandomFileName()))
+        { }
+
+        public TempDirectory(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentException("Path must be specified", nameof(path));
+            }
+
+            Directory.CreateDirectory(path);
+            this.Path = path;
+        }
+
+        ~TempDirectory()
```

Comment:
Do we need to implement finalizers?

Reply:
When using PuppeteerSharp in a server context, for example to generate PDF from HTML, my preference would be to err on the safe side and make every effort to clean up resources.